modules/core/src/main/java/org/apache/ignite/internal/util/offheap/unsafe/GridUnsafeLru.java

```diff
@@ -156,7 +156,7 @@ long entry(short order, long qAddr) {
      * @throws GridOffHeapOutOfMemoryException If failed.
      */
     long offer(int part, long addr, int hash) throws GridOffHeapOutOfMemoryException {
-        return lrus[addIdx.getAndIncrement() % cnt].offer(part, addr, hash);
+        return lrus[U.safeAbs(addIdx.getAndIncrement()) % cnt].offer(part, addr, hash);
```

Comment:
there will add U.safeAbs!
long prePoll() {
 -        int idx = rmvIdx.getAndIncrement();		 +        int idx = U.safeAbs(rmvIdx.getAndIncrement());
  		  
          // Must try to poll from each LRU.		          // Must try to poll from each LRU.
          for (int i = 0; i < lrus.length; i++) {		          for (int i = 0; i < lrus.length; i++) {
 @@ -625,4 +625,4 @@ private void entry(long qAddr, long addr) {
              return S.toString(LruStripe.class, this);		              return S.toString(LruStripe.class, this);
          }		          }
      }		      }